File: api/MyCash.API/Models/Goal.cs
using System.ComponentModel.DataAnnotations;

namespace MyCash.API.Models;

public class Goal
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid UserId { get; set; }

    [Required, MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(300)]
    public string? Description { get; set; }

    public decimal TargetAmount { get; set; }

    public decimal AllocatedAmount { get; set; }

    public DateTime? Deadline { get; set; }

    [MaxLength(20)]
    public string Color { get; set; } = "#3b82f6";

    [MaxLength(50)]
    public string Icon { get; set; } = "Target";

    public bool IsCompleted { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: web/src/types/goal.ts
export interface Goal {
  id: string
  name: string
  description?: string
  targetAmount: number
  allocatedAmount: number
  deadline?: string
  color: string
  icon: string
  isCompleted: boolean
  createdAt: string
}

export interface CreateGoalPayload {
  name: string
  description?: string
  targetAmount: number
  initialAllocation: number
  deadline?: string
  color: string
  icon: string
}

export interface UpdateGoalPayload {
  name: string
  description?: string
  targetAmount: number
  deadline?: string
  color: string
  icon: string
  isCompleted: boolean
}
